src/sinks/datadog/logs/sink.rs

```diff
@@ -140,9 +140,20 @@ impl crate::sinks::util::encoding::Encoder<Vec<Event>> for JsonEncoding {
     fn encode_input(&self, mut input: Vec<Event>, writer: &mut dyn io::Write) -> io::Result<usize> {
         for event in input.iter_mut() {
             let log = event.as_mut_log();
-            log.rename_key(self.log_schema.message_key(), event_path!("message"));
-            log.rename_key(self.log_schema.host_key(), event_path!("host"));
-            if let Some(Value::Timestamp(ts)) = log.remove(self.log_schema.timestamp_key()) {
+            let message_path = log
+                .message_path()
+                .expect("message is required (make sure the \"message\" semantic meaning is set)");
```

Comment:
Do we actually want to panic here, could there be times where the log doesn't have a "message" path?

Same for host.

Reply:
This was changed to a panic based on the conversation here: https://github.com/vectordotdev/vector/pull/15473#discussion_r1041368995

Reply:
Oh right, that sounds familiar now 😅